pkg/apis/config/defaults.go

```diff
@@ -234,3 +200,58 @@ func (d Defaults) UserContainerName(ctx context.Context) string {
 func (d Defaults) InitContainerName(ctx context.Context) string {
 	return containerNameFromTemplate(ctx, d.InitContainerNameTemplate)
 }
+
+func asTemplate(key string, target **ObjectMetaTemplate) cm.ParseFunc {
+	return func(data map[string]string) error {
+		if raw, ok := data[key]; ok {
+			val, err := template.New("container-template").Parse(raw)
+			if err != nil {
+				return err
+			}
+
+			*target = &ObjectMetaTemplate{val}
+		}
+
+		return nil
+	}
+}
+
+// ObjectMetaTemplate is a template.Template wrapper for templates
+// which can be applied to ObjectMeta. It implements DeepCopy and
+// Equal so it can be used in Config structs and with go-cmp in
+// tests.
+type ObjectMetaTemplate struct {
+	*template.Template
+}
```

Comment:
If you don't plan on adding any other fields then maybe consider making this a new type def

```suggestion
type ObjectMetaTemplate template.Template
```

Reply:
if we do that the methods of Template won't be exposed, so I think we'll either have to expose Execute on ObjectMetaTemplate with an extra wrapper function or do lots of casts everywhere. I'm not convinced that's better than just using a wrapper struct here?

Reply:
I had to do a double take on the go spec to confirm but yeah it doesn't inherit the methods - so lets leave it as is